gobblin-utility/src/main/java/gobblin/util/HadoopUtils.java

```diff
@@ -52,6 +54,21 @@
 
   public static final String HDFS_ILLEGAL_TOKEN_REGEX = "[\\s:\\\\]";
 
+  /**
+   * A {@link Collection} of all S3 schemes.
+   *
+   * <p>
+   *   The following important properties are useful to remember when writing code that is compatible with S3:
+   *   <ul>
+   *     <li>Renames are not atomic, and require copying the entire source file to the destination file</li>
+   *     <li>Writes to S3 will first go to the local filesystem, when the stream is closed the local file will be
+   *     uploaded to S3</li>
+   *   </ul>
+   * </p>
+   */
+  public static final Collection<String> S3_SCHEMES = ImmutableSortedSet.orderedBy(
```

Comment:
I agree this is the approach we probably want to go with. I will create a follow up issue for this, but its going to require a little more effort, so lets do it in a different PR. I think the work in this PR will help do some of the initial work for this also.